async/conduit_async.ml

```diff
@@ -110,7 +110,7 @@ type server = [
 
 let serve
       ?max_connections ?backlog
-      ?buffer_age_limit ?on_handler_error mode where_to_listen handle_request =
+      ?buffer_age_limit ?(on_handler_error=`Ignore) mode where_to_listen handle_request =
```

Comment:
I didn't actually think too hard about whether this is the correct default. The options are:

    -> on_handler_error       : [ `Raise
                                | `Ignore
                                | `Call of ('address -> exn -> unit)
                                ]

The default used to be `` `Raise``. I vaguely recall that this turned out to be a bad default because it meant that any random process connecting to your port and talking the wrong protocol might take down your server, so it's easy to DoS you even accidentally. But `` `Ignore`` is clearly also kind of a bad default because it's easy to ignore errors. Perhaps the right thing to do is mirror `Tcp.Server.create` and make the argument mandatory. But mostly I just wanted to get it to compile for my own project, and didn't personally fancy fixing any downstream clients :)

Reply:
relevant async_extra commit: bb08ddacc1ab6c413a58ffbd8667292d2cd82e26

I guess this should be `` `Raise`` so that it's at least behaviour-preserving. I'll go make that change.